pkg/template/registry/templateinstance/strategy.go

```diff
@@ -55,9 +55,19 @@ func (templateInstanceStrategy) PrepareForCreate(ctx apirequest.Context, obj run
 	templateInstance := obj.(*templateapi.TemplateInstance)
 
 	if templateInstance.Spec.Requester == nil {
-		user, _ := apirequest.UserFrom(ctx)
-		templateInstance.Spec.Requester = &templateapi.TemplateInstanceRequester{
-			Username: user.GetName(),
+		user, ok := apirequest.UserFrom(ctx)
+		if ok {
```

Comment:
follow the example at https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/certificates/certificates/strategy.go#L53-L77

ensure that if no user info is present, the fields are empty. the controller should refuse to act on a resource that is missing info about the requesting user